File: 0567-permutation-in-string/0567-permutation-in-string.java
class Solution {
    public boolean checkInclusion(String s1, String s2) {
        
        int[] cnt=new int[26];
        for(char c:s1.toCharArray()){
            cnt[c-'a']++;
        }
        int[]temp=new int[26];int n=s1.length();
        for(int i=0;i<s2.length();i++){
            temp[s2.charAt(i)-'a']++;
            if(i>=n){
                temp[s2.charAt(i-n)-'a']--;
            }
            if(Arrays.equals(cnt,temp))return true;
        }
        
        return false;
    }
}

File: 0567-permutation-in-string/0567-permutation-in-string.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        vector<int>s(26,0),res(26,0);
       for(char c:s1){
           s[c-'a']++;
       }
    
    for(int i=0;i<s2.size();i++){
        res[s2[i]-'a']++;
        if(i>=s1.size()){
            res[s2[i-s1.size()]-'a']--;
        }    if(s==res)return true;
        
    }
    return false;
    
    
    
    }
};